nncf/experimental/post_training/algorithms/quantization/fast_bias_correction.py

```diff
@@ -0,0 +1,225 @@
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+# pylint: disable=protected-access
+
+
+class FastBiasCorrectionParameters(AlgorithmParameters):
+    """
+    Base class of FastBiasCorrection parameters.
+    """
+
+    def __init__(self, number_samples: int = 100, threshold: float = 2.0):
+        self.number_samples = number_samples
+        self.threshold = threshold
+
+    def to_json(self) -> Dict[str, Union[str, float, int]]:
+        """
+        Serialize all FastBiasCorrection parameters to JSON.
+        """
+
+
+class FastBiasCorrection(Algorithm):
+
+    def __init__(self, parameters: FastBiasCorrectionParameters):
+        super().__init__()
+        self.number_samples = parameters.number_samples
+        self.threshold = parameters.threshold
+        self.nncf_graph = None
+        self._target_points_to_correct = []  # type: List[TargetPoint]
+        self._channel_axis_by_types = {
+            'Conv': 1,
+            'Gemm': -1,
+        }
+
+    def _generate_stat_collector(self, model_backend, axes) -> TensorStatisticCollectorBase:
+        return PTQMeanStatisticCollectorFactory.create(
+            model_backend,
+            reduction_shape=axes,
+            num_samples=self.number_samples)
+
+    @staticmethod
+    def _input_filter_func(point):
+        return PostTrainingAlgorithms.FastBiasCorrection in point.algorithm_to_tensor_collectors and \
+            point.target_point.type == TargetType.PRE_LAYER_OPERATION
+
+    @staticmethod
+    def _output_filter_func(point):
+        return PostTrainingAlgorithms.FastBiasCorrection in point.algorithm_to_tensor_collectors and \
+            point.target_point.type == TargetType.POST_LAYER_OPERATION
+
+    def _apply(self, model: ModelType, engine: Engine,
+               statistic_points: StatisticPointsContainer) -> ModelType:
+
+        model_backend = infer_backend_from_model(model)
+
+        transformation_layout = TransformationLayout()
+        nncf_graph = NNCFGraphFactory.create(model)
+
+        layers_with_bias_types = PTQLayerMetatypesFactory.get_layers_with_bias_types(
+            model_backend)
+        biased_nodes = nncf_graph.get_nodes_by_metatypes(
+            layers_with_bias_types)
+
+        for node in biased_nodes:
+            node_name = node.node_name
+            input_fp = []
+            input_shape = []
+            output_fp = []
+
+            for tensor_collector in statistic_points.iter_through_algorithm_tensor_collectors_in_target_node(
+                    node_name,
+                    self._input_filter_func,
+                    PostTrainingAlgorithms.FastBiasCorrection):
+                input_fp.extend(tensor_collector.get_statistics().mean_values)
+                input_shape.extend(tensor_collector.get_statistics().shape)
+            for tensor_collector in statistic_points.iter_through_algorithm_tensor_collectors_in_target_node(
```

Comment:
I would split the code a little bit just for future generations), like:
- get_fp_inputs()
- get_fp_outputs()
- extract_submodel()
- get_bias_shift()
- correct_bias()

Reply:
Done.